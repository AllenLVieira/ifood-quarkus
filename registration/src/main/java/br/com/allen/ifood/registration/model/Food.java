package br.com.allen.ifood.registration.model;

import io.quarkus.hibernate.orm.panache.PanacheEntityBase;

import javax.persistence.*;
import java.math.BigDecimal;

@Entity
@Table(name = "food")
public class Food extends PanacheEntityBase {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    public Long id;

    public String name;

    public String description;

    @ManyToOne
    public Restaurant restaurant;

    public BigDecimal price;
}
