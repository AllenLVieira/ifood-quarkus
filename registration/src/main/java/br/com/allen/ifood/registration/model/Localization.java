package br.com.allen.ifood.registration.model;

import io.quarkus.hibernate.orm.panache.PanacheEntityBase;

import javax.persistence.*;

@Entity
@Table(name = "localization")
public class Localization extends PanacheEntityBase {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    public Long id;

    public Double latitude;

    public Double longitude;
}
