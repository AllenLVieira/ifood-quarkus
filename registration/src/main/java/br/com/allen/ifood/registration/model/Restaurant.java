package br.com.allen.ifood.registration.model;

import io.quarkus.hibernate.orm.panache.PanacheEntityBase;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import javax.persistence.*;
import java.util.Date;

@Entity
@Table(name = "restaurant")
public class Restaurant  extends PanacheEntityBase {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    public Long id;

    public String owner;

    public String cnpj;

    public String name;

    @ManyToOne
    public Localization Localization;

    @CreationTimestamp
    public Date creationDate;

    @UpdateTimestamp
    public Date updateDate;
}
