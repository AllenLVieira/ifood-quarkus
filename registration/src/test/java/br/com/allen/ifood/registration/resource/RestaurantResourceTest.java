package br.com.allen.ifood.registration.resource;

import br.com.allen.ifood.registration.RegistrationTestLifecycleManager;
import br.com.allen.ifood.registration.model.Restaurant;
import com.github.database.rider.cdi.api.DBRider;
import com.github.database.rider.core.api.configuration.DBUnit;
import com.github.database.rider.core.api.configuration.Orthography;
import com.github.database.rider.core.api.dataset.DataSet;
import io.quarkus.test.common.QuarkusTestResource;
import io.quarkus.test.junit.QuarkusTest;
import io.restassured.RestAssured;
import io.restassured.http.ContentType;
import io.restassured.specification.RequestSpecification;
import org.approvaltests.Approvals;
import org.junit.Assert;
import org.junit.jupiter.api.Test;

import javax.ws.rs.core.Response;

import static io.restassured.RestAssured.given;

@QuarkusTest
@QuarkusTestResource(RegistrationTestLifecycleManager.class)
@DBRider
@DBUnit(caseInsensitiveStrategy = Orthography.LOWERCASE)
class RestaurantResourceTest {

    @Test
    @DataSet("restaurants-first-scenario-1.yml")
    void getAllRestaurants() {
        String result = given()
                .when().get("/restaurants")
                .then()
                .statusCode(200)
                .extract().asString();
        Approvals.verifyJson(result);
    }

    // Corrigir retorno 415
    private RequestSpecification given() {
        return RestAssured.given().contentType(ContentType.JSON);
    }

    @Test
    @DataSet("restaurants-first-scenario-1.yml")
    public void updateRestaurant() {
        Restaurant dto = new Restaurant();
        dto.name = "newName";
        Long parameterValue = 1L;
        String result = given()
                .with().pathParam("restaurantId", parameterValue)
                .body(dto)
                .when().put("/restaurants/{restaurantId}")
                .then()
                .statusCode(Response.Status.NO_CONTENT.getStatusCode())
                .extract().asString();
        Assert.assertEquals("", result);
        Restaurant findById = Restaurant.findById(parameterValue);
        Assert.assertEquals(dto.name, findById.name);

    }
}