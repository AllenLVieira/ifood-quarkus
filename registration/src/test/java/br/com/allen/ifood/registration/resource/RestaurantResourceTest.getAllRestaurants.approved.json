[
  {
    "id": 1,
    "name": "New restaurant",
    "owner": "Owner"
  },
  {
    "id": 2,
    "name": "New restaurant2",
    "owner": "Owner2"
  }
]